eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWallet.scala

```diff
@@ -90,7 +90,13 @@ class BitcoinCoreWallet(rpcClient: BitcoinJsonRPCClient)(implicit ec: ExecutionC
     future.map(_.forall(b => b))
   }
 
-  override def getBalance: Future[Satoshi] = rpcClient.invoke("getbalance") collect { case JDecimal(balance) => Satoshi(balance.bigDecimal.scaleByPowerOfTen(8).longValue) }
+  override def getBalance: Future[OnChainBalance] = rpcClient.invoke("getbalances").map(json => {
+    def toSatoshi(v: BigDecimal): Satoshi = Satoshi(v.bigDecimal.scaleByPowerOfTen(8).longValue)
```

Comment:
Those decimal/integer conversions are tricky. I think the formula is correct, but would a few unit tests be over the top?

Reply:
I agree, it's probably worth adding a unit test, it never hurts.

Reply:
I improved the test in https://github.com/ACINQ/eclair/pull/1461/commits/dcb67246eaf0e53f7cdde1346e74f92e4d13204f, if the conversion was incorrect this test wouldn't pass.